vnext/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +15,268 @@ enum class ScrollViewCommands
+      if (valid)
+      {
+        scrollViewer.ChangeView(nullptr, nullptr, static_cast<float>(zoomScale));
+      }
+    }
+    else if (propertyName == "snapToInterval")
+    {
+      const auto [valid, snapToInterval] = getPropertyAndValidity(propertyValue, 0.0);
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToInterval(static_cast<float>(snapToInterval));
+      }
+    }
+    else if (propertyName == "snapToOffsets")
+    {
+      const auto [valid, snapToOffsets] = getPropertyAndValidity(propertyValue, winrt::single_threaded_vector<float>());
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToOffsets(snapToOffsets.GetView());
+      }
+    }
+    else if (propertyName == "snapToAlignment")
+    {
+      const auto [valid, snapToAlignment] = getPropertyAndValidity(propertyValue, winrt::SnapPointsAlignment::Near);
+      if (valid)
+      {
+        scrollViewer.HorizontalSnapPointsAlignment(snapToAlignment);
+        scrollViewer.VerticalSnapPointsAlignment(snapToAlignment);
+      }
+    }
+  }
+
+  Super::updateProperties(std::move(reactDiffMap));
+  m_updating = false;
+}
+
+template <typename T>
+std::tuple<bool, T> ScrollViewShadowNode::getPropertyAndValidity(folly::dynamic propertyValue, T defaultValue)
+{
+  if (propertyValue.isNull())
+  {
+    return std::make_tuple(true, defaultValue);
+  }
+
+  if constexpr (std::is_same<T, bool>())
+  {
+    if (propertyValue.isBool())
+    {
+      return std::make_tuple(true, propertyValue.asBool());
+    }
+  }
+
+  if constexpr (std::is_same<T, double>())
+  {
+    if (propertyValue.isDouble())
+    {
+      return std::make_tuple(true, static_cast<T>(propertyValue.asDouble()));
+    }
+    if (propertyValue.isInt())
+    {
+      return std::make_tuple(true, static_cast<T>(propertyValue.asInt()));
+    }
+  }
+
+  if constexpr (std::is_same<T, winrt::IVector<float>>())
+  {
+    if (propertyValue.isArray())
+    {
+      return std::make_tuple(
+        true,
+        [propertyValue]()
+        {
+          const auto vector = winrt::single_threaded_vector<float>();
+          for (const auto val : propertyValue)
+          {
+            if (val.isNumber())
+              vector.Append(static_cast<float>(val.asDouble()));
+          }
+          return vector;
+        }());
+    }
+  }
+
+  if constexpr (std::is_same<T, winrt::SnapPointsAlignment>())
+  {
+    if (propertyValue.isString())
+    {
+      return std::make_tuple(
+        true,
+        static_cast<T>([snapToAlignment = propertyValue.asString()]()
+      {
+        if (snapToAlignment == "end")
+        {
+          return winrt::SnapPointsAlignment::Far;
+        }
+        else if (snapToAlignment == "center")
+        {
+          return winrt::SnapPointsAlignment::Center;
+        }
+        else
+        {
+          return winrt::SnapPointsAlignment::Near;
+        }
+      }()));
+    }
```

Comment:
is this part used from multiple callers?  I could see some of the others benefiting